core/unit_test/hip/TestHIP_Spaces.cpp

```diff
@@ -0,0 +1,232 @@
+                    Kokkos::Experimental::HIPSpace>::accessible,
+                "");
+
+  //--------------------------------------
+
+  static_assert(
+      !Kokkos::Impl::SpaceAccessibility<Kokkos::Experimental::HIP,
+                                        Kokkos::HostSpace>::accessible,
+      "");
+
+  static_assert(Kokkos::Impl::SpaceAccessibility<
+                    Kokkos::Experimental::HIP,
+                    Kokkos::Experimental::HIPSpace>::accessible,
+                "");
+
+  static_assert(Kokkos::Impl::SpaceAccessibility<
+                    Kokkos::Experimental::HIP,
+                    Kokkos::Experimental::HIPHostPinnedSpace>::accessible,
+                "");
+
+  static_assert(
+      !Kokkos::Impl::SpaceAccessibility<
+          Kokkos::HostSpace, Kokkos::Experimental::HIPSpace>::accessible,
+      "");
+
+  static_assert(Kokkos::Impl::SpaceAccessibility<
+                    Kokkos::HostSpace,
+                    Kokkos::Experimental::HIPHostPinnedSpace>::accessible,
+                "");
+
+  static_assert(
+      std::is_same<
+          Kokkos::Impl::HostMirror<Kokkos::Experimental::HIPSpace>::Space,
+          Kokkos::HostSpace>::value,
+      "");
+
+  static_assert(
+      std::is_same<Kokkos::Impl::HostMirror<
+                       Kokkos::Experimental::HIPHostPinnedSpace>::Space,
+                   Kokkos::Experimental::HIPHostPinnedSpace>::value,
+      "");
+
+  static_assert(Kokkos::Impl::SpaceAccessibility<
+                    Kokkos::Impl::HostMirror<Kokkos::Experimental::HIP>::Space,
+                    Kokkos::HostSpace>::accessible,
+                "");
+
+  static_assert(
+      Kokkos::Impl::SpaceAccessibility<
+          Kokkos::Impl::HostMirror<Kokkos::Experimental::HIPSpace>::Space,
+          Kokkos::HostSpace>::accessible,
+      "");
+
+  static_assert(Kokkos::Impl::SpaceAccessibility<
+                    Kokkos::Impl::HostMirror<
+                        Kokkos::Experimental::HIPHostPinnedSpace>::Space,
+                    Kokkos::HostSpace>::accessible,
+                "");
+}
+
+template <class MemSpace, class ExecSpace>
+struct TestViewHIPAccessible {
+  enum { N = 1000 };
+
+  using V = Kokkos::View<double *, MemSpace>;
+
+  V m_base;
+
+  struct TagInit {};
+  struct TagTest {};
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const TagInit &, const int i) const { m_base[i] = i + 1; }
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const TagTest &, const int i, long &error_count) const {
+    if (m_base[i] != i + 1) ++error_count;
+  }
+
+  TestViewHIPAccessible() : m_base("base", N) {}
+
+  static void run() {
+    TestViewHIPAccessible self;
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy<typename MemSpace::execution_space, TagInit>(0, N),
+        self);
+    typename MemSpace::execution_space().fence();
+
+    // Next access is a different execution space, must complete prior kernel.
+    long error_count = -1;
```

Comment:
Initialize to `-1`?

Reply:
What's the problem?

Reply:
Just surprised but I saw that the Cuda implementation does that as well.